x/programs/cmd/simulator/cmd/plan.go

```diff
@@ -4,86 +4,108 @@
 package cmd
 
 import (
+	"bufio"
 	"context"
 	"crypto/rand"
 	"errors"
 	"fmt"
-	"io"
 	"math"
 	"os"
 	"strconv"
 	"strings"
 	"time"
 
-	"github.com/ava-labs/hypersdk/x/programs/cmd/simulator/vm/actions"
-
-	"github.com/ava-labs/hypersdk/x/programs/program"
-
-	"github.com/spf13/cobra"
+	"github.com/akamensky/argparse"
 	"go.uber.org/zap"
 
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/utils/logging"
 
 	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/x/programs/cmd/simulator/vm/actions"
 	"github.com/ava-labs/hypersdk/x/programs/cmd/simulator/vm/storage"
 	"github.com/ava-labs/hypersdk/x/programs/cmd/simulator/vm/utils"
+	"github.com/ava-labs/hypersdk/x/programs/program"
 )
 
+var _ Cmd = (*runCmd)(nil)
+
 type runCmd struct {
-	plan *Plan
-	log  logging.Logger
-	db   *state.SimpleMutable
+	cmd *argparse.Command
+
+	stdin    *bool
+	lastStep *int
+	file     *string
+
+	step   *Step
+	log    logging.Logger
+	db     **state.SimpleMutable
+	reader **bufio.Reader
 
 	// tracks program IDs created during this simulation
 	programIDStrMap map[string]string
-	stdinReader     io.Reader
 }
 
-func newRunCmd(log logging.Logger, db *state.SimpleMutable) *cobra.Command {
-	r := &runCmd{
-		log:             log,
-		db:              db,
-		programIDStrMap: make(map[string]string),
+func (c *runCmd) New(parser *argparse.Parser, db **state.SimpleMutable, programIDStrMap map[string]string, lastStep *int, reader **bufio.Reader) {
+	c.db = db
+	c.programIDStrMap = programIDStrMap
+	c.cmd = parser.NewCommand("run", "Run a HyperSDK program simulation plan")
+	c.stdin = c.cmd.Flag("", "stdin", &argparse.Options{
+		Help:     "name of the key to use to deploy the program",
+		Required: false,
+		Default:  false,
+	})
+	c.file = c.cmd.String("", "file", &argparse.Options{
+		Required: false,
+	})
+	c.lastStep = lastStep
+	c.reader = reader
+}
+
+func (c *runCmd) Run(ctx context.Context, log logging.Logger, args []string) (*Response, error) {
+	c.log = log
+	resp1 := newResponse(0)
+	resp1.setTimestamp(time.Now().Unix())
+	err := resp1.Print()
+	if err != nil {
+		return newResponse(0), err
```

Comment:
Same as above about the ID of 0 by default. We also return a "default" response with a missing timestamp in case of an error because this response will not be printed anyway, is that the correct way to return a kind of "nil" reponse ?

Reply:
That's a good question. I'm inclined to think that we should just print to stderr and exit with a non-0 exit code. Maybe there's a case where we should handle the error though...

Reply:
@dboehm-avalabs Is that fine to use a partially incorrect `Response` type since it won't be used anyway as a "default" ?